Copy number variation (CNV)
```bash
wget https://gdc.xenahubs.net/download/GDC-PANCAN.masked_cnv.tsv.gz
gzip -d GDC-PANCAN.masked_cnv.tsv.gz
```
Raw data was generated by Affymetrix SNP 6.0 array.
The CNV files are genomic regions and log2(copy-number /2). Diploid regions will have a segment mean of zero, amplified regions will have positive values, and deletions will have negative values.
Copy Number Estimation
-   Genes with focal CNV values smaller than -0.3 are categorized as a "loss" (-1)
-   Genes with focal CNV values larger than 0.3 are categorized as a "gain" (+1)
-   Genes with focal CNV values between and including -0.3 and 0.3 are categorized as "neutral" (0).

```bash
qlogin -l h_vmem=8G
module load igmm/apps/R/3.6.3
cd /exports/eddie/scratch/s1949868/CopyNumber
R --no-restore
```
```r
# read PRDM9 expression
PRDM9.expression <- read.delim("/home/s1949868/MScProject/Results/PRDM9ExpressionAndBinding/PRDM9Expression.txt", sep = "\t",header = TRUE)
# store sampleID in a new column
PRDM9.expression$sampleID <- substr(rownames(PRDM9.expression),6,21)

# read cnv
masked_cnv <- read.delim("GDC-PANCAN.masked_cnv.tsv",sep = "\t",header = TRUE)
masked_cnv$sample <- as.character(masked_cnv$sample)
masked_cnv$Chrom <- as.character(masked_cnv$Chrom)
masked_cnv$Chrom <- paste0("chr",masked_cnv$Chrom)

# select segments considered as gain or loss
masked_cnv <- masked_cnv[masked_cnv$value >= 0.3 | masked_cnv$value <= -0.3,]
# take the breakpoints
masked_cnv_start <- masked_cnv[,c(1,2,3,5)]
colnames(masked_cnv_start) <- c("sample", "Chrom", "End", "value")
masked_cnv_end <- masked_cnv[,c(1,2,4,5)]
masked_cnv_bp <- rbind(masked_cnv_start, masked_cnv_end)
masked_cnv_bp$Start <- masked_cnv_bp$End - 1
masked_cnv_breakpoints <- data.frame("sample" = masked_cnv_bp$sample, "Chrom" = masked_cnv_bp$Chrom, "Start" = masked_cnv_bp$Start, "End" = masked_cnv_bp$End, "value" = masked_cnv_bp$value, stringsAsFactors = FALSE)

# just keep samples containing RNA-seq, ATAC-seq and cnv data.
masked_cnv_breakpoints <- masked_cnv_breakpoints[masked_cnv_breakpoints$sample %in% PRDM9.expression$sampleID,]

# save breakpoints of each sample separately
for (i in unique(masked_cnv_breakpoints$sample)) {
  output <- masked_cnv_breakpoints[masked_cnv_breakpoints$sample == i,c(2,3,4,5)]
  cType_sampleID <- rownames(PRDM9.expression[PRDM9.expression$sampleID == i,])
  
  write.table(output,
	file=paste0(cType_sampleID,".masked_cnv_breakpoints.txt"),
	sep = "\t",
	append=FALSE,row.names = FALSE,col.names = FALSE,
	quote =FALSE
	)
}
```
```bash
module load igmm/apps/BEDTools/2.27.1

for bpFile in $(ls /exports/eddie/scratch/s1949868/CopyNumber/SVB/*breakpoints.txt); do 
	echo $bpFile; 
	sampleID=${bpFile#*SVB/}; 
	sampleID=${sampleID%.masked_cnv*};
	echo $sampleID
	
	bedtools slop -i ${bpFile} -g /home/s1949868/Tools/chr_length.hg38.txt -b 100 > ${sampleID}.SVBS.txt
	done
```
```bash


echo -e "sampleID\tnumSVBs\tnumOverlap\toverlapFraction" > overlap_PRDM9BoundPeaks_VS_SVBs.txt

for SVBSFile in $(ls /exports/eddie/scratch/s1949868/CopyNumber/SVBS/*SVBS.txt); do 
	echo $SVBSFile; 
	sampleID=${SVBSFile#*SVBS/}; 
	sampleID=${sampleID%.SVBS*};
	echo $sampleID
	
	motifFile=/exports/eddie/scratch/s1949868/MotifFind_fimo_404/allFimoGFF_CaseID/${sampleID}_peakCalls_fimo.gff
	echo $PRDM9BoundPeaks
	
	a=`cat $bpFile | wc -l`
	b=`bedtools intersect -a $bpFile -b $PRDM9BoundPeaks -f 1.0 -u | wc -l`
	
	c=`echo "scale=2;$b/$a" | bc`
	echo -e "$sampleID\t$a\t$b\t$c" >> overlap_PRDM9BoundPeaks_VS_SVBs.txt
done



```
<!--stackedit_data:
eyJoaXN0b3J5IjpbLTE5ODEzMDE3OTMsMTQxMjkzNzc4LDgxMz
I3NTE5NywtMTI5NTI1NDg5MSw4MTY5MzAwNzEsLTcwNTc1MTA1
MSw2Nzg0MzAxNjQsMTQ5MTQ2MDQ3MywtNjQ1ODc2MDcwLDE5ND
gxODY4NjYsLTE2MTk4NTMwNzMsMzk1OTk3MTg2LDEzNTUwMDky
OTksMzAxNTc2OTkxLDEyNzc0MDY4MDIsNDE2MjczMjI0LDEzNj
I0OTE5MDksLTk1OTg3NDkyNiwyOTE2NzE2NzUsOTQ4NzgzNzBd
fQ==
-->