Copy number variation (CNV)
```bash
wget https://gdc.xenahubs.net/download/GDC-PANCAN.masked_cnv.tsv.gz
gzip -d GDC-PANCAN.masked_cnv.tsv.gz
```
Raw data was generated by Affymetrix SNP 6.0 array.
The CNV files are genomic regions and log2(copy-number /2). Diploid regions will have a segment mean of zero, amplified regions will have positive values, and deletions will have negative values.
Copy Number Estimation
-   Genes with focal CNV values smaller than -0.3 are categorized as a "loss" (-1)
-   Genes with focal CNV values larger than 0.3 are categorized as a "gain" (+1)
-   Genes with focal CNV values between and including -0.3 and 0.3 are categorized as "neutral" (0).

```bash
qlogin -l h_vmem=8G
module load igmm/apps/R/3.6.3

PRDM9.expression <- read.delim("/home/s1949868/Results/PRDM9ExpressionAndBinding/PRDM9Expression.txt", sep = "\t",header = TRUE)

masked_cnv <- read.delim("~/GDC-PANCAN.masked_cnv.tsv",sep = "\t",header = TRUE)
masked_cnv$sample <- as.character(masked_cnv$sample)
masked_cnv$Chrom <- as.character(masked_cnv$Chrom)
masked_cnv$Chrom <- paste0("chr",masked_cnv$Chrom)


masked_cnv[masked_cnv$sample %in% substr(rownames(PRDM9.expression),6,21),]
saveRDS(object = masked_cnv, file = "masked_cnv.rds")

masked_cnv$Start <- masked_cnv$Start-100
masked_cnv$End <- masked_cnv$End+100
```

<!--stackedit_data:
eyJoaXN0b3J5IjpbMjExNDc0Nzk1OCw0ODM1NjM2MjYsLTE2OT
YzODkxMzIsLTExMjEyNDE0OTgsMTUwNjMyODM4Ml19
-->