Copy number variation (CNV)
```bash
wget https://gdc.xenahubs.net/download/GDC-PANCAN.masked_cnv.tsv.gz
gzip -d GDC-PANCAN.masked_cnv.tsv.gz
```
Raw data was generated by Affymetrix SNP 6.0 array.
The CNV files are genomic regions and log2(copy-number /2). Diploid regions will have a segment mean of zero, amplified regions will have positive values, and deletions will have negative values.
Copy Number Estimation
-   Genes with focal CNV values smaller than -0.3 are categorized as a "loss" (-1)
-   Genes with focal CNV values larger than 0.3 are categorized as a "gain" (+1)
-   Genes with focal CNV values between and including -0.3 and 0.3 are categorized as "neutral" (0).

```bash
qlogin -l h_vmem=8G
module load igmm/apps/R/3.6.3
cd /exports/eddie/scratch/s1949868/CopyNumber
R
```
```r
# read PRDM9 expression
PRDM9.expression <- read.delim("/home/s1949868/Results/PRDM9ExpressionAndBinding/PRDM9Expression.txt", sep = "\t",header = TRUE)
# store sampleID in a new column
PRDM9.expression$sampleID <- substr(rownames(PRDM9.expression),6,21)

# read cnv
masked_cnv <- read.delim("GDC-PANCAN.masked_cnv.tsv",sep = "\t",header = TRUE)
masked_cnv$sample <- as.character(masked_cnv$sample)
masked_cnv$Chrom <- as.character(masked_cnv$Chrom)
masked_cnv$Chrom <- paste0("chr",masked_cnv$Chrom)

# just keep samples containing RNA-seq, ATAC-seq and cnv data.
masked_cnv <- masked_cnv[masked_cnv$sample %in% PRDM9.expression$sampleID,]

for (i in unique(masked_cnv$sample)) {
  output <- masked_cnv[masked_cnv$sample == i,c(2,3,4,5)]
  cType_sampleID <- rownames(PRDM9.expression[PRDM9.expression$sampleID == i,])
  
  write.table(output,
	file=paste0(cType_sampleID,".masked_cnv.txt"),
	sep = "\t",
	append=FALSE,row.names = FALSE,col.names = TRUE,
	quote =FALSE
	)
}
```
```bash

```
<!--stackedit_data:
eyJoaXN0b3J5IjpbLTE3NDc5NjY3MSwtNTIzMjg0NjYzLC0xNj
k5MDEyMjgxLDIwNzQwNzE3LC0xMzI3MTgyMDk3LDQ4MzU2MzYy
NiwtMTY5NjM4OTEzMiwtMTEyMTI0MTQ5OCwxNTA2MzI4MzgyXX
0=
-->